File: 169/169.py
from typing import List

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        idx, c = 0, 1
        for i in range(1, len(nums)):
            if nums[idx]==nums[i]:
                c += 1
            else:
                c -= 1
                if c < 0:
                    idx = i
                    c = 1
        return nums[idx]

File: leetcode/169/169.go
package main

func majorityElement(nums []int) int {
	idx, cnt := 0, 1
	for i := 1; i < len(nums); i++ {
		if nums[idx] == nums[i] {
			cnt++
		} else {
			cnt--
			if cnt == 0 {
				idx = i
				cnt = 1
			}
		}
	}
	return nums[idx]
}
